src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2703,6 +2704,19 @@ private bool HasVariantConversionNoCycleCheck(NamedTypeSymbol source, NamedTypeS
                     switch (typeParameterSymbol.Variance)
                     {
                         case VarianceKind.None:
+                            // System.IEquatable<T> is invariant for back compat reasons (dynamic type checks could start
+                            // to succeed where they previously failed, creating different runtime behavior), but the uses
+                            // require treatment specifically of nullability as contravariant, so we special case the
+                            // behavior here. Normally we use GetWellKnownType for these kinds of checks, but in this
+                            // case we don't want just the canonical IEquatable to be special-cased, we want all definitions
+                            // to be treated as contravariant, in case there are other definitions in metadata that were
```

Comment:
Is this a case we actually care about? Where would we care about it?

Reply:
The main reason why we would care is we have TypeConversions, which doesn't necessarily use a Compilation to ask about conversions between types, so trying to find a well-known type is fraught.